Menus/LoadingGUI.h

```diff
@@ -0,0 +1,53 @@
+#ifndef _LOADINGUI_
+#define _LOADINGGUI_
+
+namespace RTE {
+
+	class LoadingGUI {
+
+	public:
+
+		/// <summary>
+		/// Creates the loading screen and the log writer, then calls loading all the modules.
+		/// </summary>
+		void InitLoadingScreen();
+
+		/// <summary>
+		/// Updates the loading progress report.
+		/// </summary>
+		/// <param name="reportString">The string to print in the report and log.</param>
+		/// <param name="newItem">Whether to start a new line in the log writer and to scroll the bitmap.</param>
+		static void LoadingSplashProgressReport(std::string reportString, bool newItem = false);
+
+		/// <summary>
+		/// Finding and loading all DataModules
+		/// </summary>
+		bool LoadDataModules();
+
+		/// <summary>
+		/// Unzip all *.rte.zip files found in the install directory, overwriting all files already existing.
+		/// This will cause extracted and available data modules to be updated to whatever is within their corresponding zip files.
+		/// The point of this is that it facilitates downloaded mods being loaded without having to be manually unzipped first by the user.
+		/// </summary>
+		void ExtractArchivedModules();
+
+		/// <summary>
+		/// Sets whether to print the progress report log to command-line.
+		/// </summary>
+		/// <param name="enable">True to print progress report log to command-line.</param>
+		void SetLogToCLI(bool enable) { m_LogToCLI = true; }
+
+	protected:
+
+		static int m_LoadingGUIPosX; //! Position of the progress report box on X axis.
```

Comment:
They're static because `LoadingSplashProgressReport` is static and unless they are too squiggly lines are everywhere.
And if `LoadingSplashProgressReport` isn't static, the callback is broken and everything is on fire.